File: src/main/java/com/dude/dms/backend/service/DocHistoryService.java
package com.dude.dms.backend.service;

import com.dude.dms.backend.data.base.Doc;
import com.dude.dms.backend.data.history.DocHistory;
import com.dude.dms.backend.repositories.DocHistoryRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class DocHistoryService extends HistoryCrudService<Doc, DocHistory> {

    private final DocHistoryRepository docHistoryRepository;

    @Autowired
    public DocHistoryService(DocHistoryRepository docHistoryRepository) {
        this.docHistoryRepository = docHistoryRepository;
    }

    @Override
    public DocHistoryRepository getRepository() {
        return docHistoryRepository;
    }

    @Override
    public List<DocHistory> getHistory(Doc entity) {
        return docHistoryRepository.findByDoc(entity);
    }
}

File: src/main/java/com/dude/dms/backend/service/DocHistoryService.kt
package com.dude.dms.backend.service

import com.dude.dms.backend.data.docs.Doc
import com.dude.dms.backend.data.history.DocHistory
import com.dude.dms.backend.repositories.DocHistoryRepository
import com.dude.dms.brain.events.EventManager
import org.springframework.stereotype.Service

@Service
class DocHistoryService(
        private val docHistoryRepository: DocHistoryRepository,
        eventManager: EventManager
) : HistoryCrudService<Doc, DocHistory>(docHistoryRepository, eventManager) {

    override fun getHistory(entity: Doc) = docHistoryRepository.findByDoc(entity)
}